index.js

```diff
@@ -174,11 +174,15 @@ module.exports = {
     this.initPlugins();
     this.processingTree = this.createProcessingTree();
 
-    this.usesBlurhash = this.addonOptions.images.some(
-      (imageConfig) => imageConfig.lqip && imageConfig.lqip.type === 'blurhash'
-    );
-    this.options['@embroider/macros'].setOwnConfig.usesBlurhash =
-      this.usesBlurhash;
+    this.usesBlurhash =
+      this.addonOptions.usesBlurhash ||
```

Comment:
I wonder how we should deal with a case where we have local blurhash images, but `usesBlurhash` is set to false? Currently it would still use blurhash, so setting to false has no effect (due to how `||` works). We could change this to only check the local images when it is `undefined` (basically use `??` instead of `||`, but node 12 does not support this yet, does it?)

Or should we throw a build-time assertion in that case, basically saying that this is an invalid contradiction?

Reply:
The current behavior of the addon is to automatically turn on blurhash when there are images that should use blurhash. I think that's the correct behavior and I doubt anybody would want to turn it off blurhash when they have images that are defined to use blurhash.

The PR only makes sure that you can opt-in to use blurhash when there are no pre-build images that are defined to use blurhash - e.g. an app that I'm working on wants to use blurhash in an image search but has no images to precompile.

I personally feel that allowing users to opt-into including the blurhash library is useful - allowing them to disable it although they apparently have images that should use that mechanism is a potential troll that we might want to avoid?!

Reply:
> allowing them to disable it although they apparently have images that should use that mechanism is a potential troll that we might want to avoid

Yes, right. That's why I thought we should maybe throw an assertion. Something like that:

```js
const imagesUseBluhash = this.addonOptions.images.some(
      (imageConfig) => imageConfig.lqip && imageConfig.lqip.type === 'blurhash'
    );
if (this.addonOptions.usesBlurhash === false && imagesUseBluhash) {
  throw new SilentError('You cannot disable blurhash support while it is still used in your images config. Please remove any instance of `lqip: { type: 'blurhash' }` from your images config.');
}  
```

Reply:
Makes sense - I pushed an update.